Frends.Community.Email.Tests/ExchangeEmailTests.cs

```diff
@@ -0,0 +1,651 @@
+                EmailSubjectFilter = subject
+            };
+
+            Directory.CreateDirectory(dirPath);
+            var result = await ReadEmailTask.ReadEmailFromExchangeServer(settings, options, new CancellationToken());
+            Assert.IsTrue(result.Count == 1);
+            Directory.Delete(dirPath, true);
+            await DeleteMessages(subject);
+        }
+
+        [Test]
+        public async Task ReadEmailFromExchangeServer_ShouldOverwriteAttachmentsTest()
+        {
+            var dirPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "../../../OverwriteData/");
+            var subject = "Overwrite Attchment Test";
+            await SendTestEmailWithAttachment(subject, "OverwriteAttachment.txt");
+            var settings = new ExchangeSettings
+            {
+                TenantId = _tenantID,
+                AppId = _applicationID,
+                Username = _username,
+                Password = _password,
+                Mailbox = _mailbox
+            };
+            var options = new ExchangeOptions
+            {
+                MaxEmails = 1,
+                DeleteReadEmails = false,
+                GetOnlyUnreadEmails = false,
+                MarkEmailsAsRead = false,
+                IgnoreAttachments = false,
+                GetOnlyEmailsWithAttachments = true,
+                AttachmentSaveDirectory = dirPath,
+                OverwriteAttachment = true
+            };
+
+            Directory.CreateDirectory(dirPath);
+            var result = await ReadEmailTask.ReadEmailFromExchangeServer(settings, options, new CancellationToken());
+            Assert.IsTrue(File.Exists(result[0].AttachmentSaveDirs[0]));
+            Assert.IsTrue(Directory.GetFiles(dirPath).Length == 1);
+            result = await ReadEmailTask.ReadEmailFromExchangeServer(settings, options, new CancellationToken());
+            Assert.IsTrue(File.Exists(result[0].AttachmentSaveDirs[0]));
+            Assert.IsTrue(Directory.GetFiles(dirPath).Length == 1);
+            Directory.Delete(dirPath, true);
+            await DeleteMessages(subject);
+        }
+
+        [Test]
+        public async Task SendEmailWithPlainTextTest()
+        {
+            var input = new ExchangeInput
+            {   
+                To = _username,
+                Message = "This is a test message from Frends.Commmunity.Email Unit Tests.",
+                IsMessageHtml = false,
+                MessageEncoding = "utf-8",
+                Subject = "Email test - PlainText"
+            };
+
+            var server = new ExchangeServer
+            {
+                Username = _username,
+                Password = _password,
+                AppId = _applicationID,
+                TenantId = _tenantID
+            };
+
+            var result = await EmailTask.SendEmailToExchangeServer(input, null, server, new CancellationToken());
+            Assert.IsTrue(result.EmailSent);
+            await DeleteMessages(input.Subject);
```

Comment:
Should we actually read the email (e.g. using read task) and check that it is actually plain text?